chore: add http2 alpn test using fastify

mkaufmaner opened the pull request.

metcoder95 commented:
@KhafraDev, is better to have a custom implementation instead of adding other dev dependencies for testing out the ALPN negotiation?

If so, I haven't tried yet but we should work on a small fixture that supports ALPN negotiation for testing.

metcoder95 commented on test/http2-alpn.js:
```diff
@@ -0,0 +1,131 @@
+
+const { Client } = require('..')
+
+test('Should support secure HTTPS connection', async (t) => {
```
This scenario is not really needed, we are just looking to test the ALPN negotiation

mkaufmaner replied:
Yep, saw your comment in the other PR and will work towards using native http/2.

mkaufmaner replied:
Looks like I can use the http2 compatibility layer for this. Will try and modify accordingly.

KhafraDev commented:
I'm sorry, I completely misunderstood what was being asked. Adding dev dependencies is fine.

mkaufmaner commented:
> I'm sorry, I completely misunderstood what was being asked. Adding dev dependencies is fine.

Haha, well I think I have already pretty much got it working without fastify anyways. Will update shortly.

mkaufmaner commented:
@KhafraDev I removed fastify as a dev dependency and I still somewhat agree that external dependencies shouldn't be added unless absolutely necessary. I ended up getting ALPN to work with barebones `http2.createSecureServer` and the `allowHTTP1: true` option.

@metacoder I added a test for handling request bodies as well using a `POST` method.

@metcoder95 This is ready to merge when you are.

metcoder95 commented on package.json:
```diff
@@ -80,6 +80,7 @@
     "delay": "^5.0.0",
     "dns-packet": "^5.4.0",
     "docsify-cli": "^4.4.3",
+    "fastify": "^4.17.0",
```
No needed anymore?

mkaufmaner commented:
@metcoder95 I believe this is ready to be merged.
